### DISCUSSION TITLE
Presets for nodes in the Create New Node dialog

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#9557] [Editor]
Many nodes are useless without some basic setup, that setup is usually a repetitive task that ends being annoying unless a lot of work is put in creating custom nodes.

The UX could be improved a lot with the use of presets when adding nodes, for example:

* Particles

To create a basic working particles emitter the user needs to add a GPUParticles node, then the node will show a warning mentioning missing properties.

For example, on 3D particles, there are 2 missing properties.
![image](https://github.com/godotengine/godot-proposals/assets/10215987/c6ad99ba-1dcb-46d3-8438-6ea2e111bbcd)

One can guess where are them but are not directly highlighted or anything. 

![image](https://github.com/godotengine/godot-proposals/assets/10215987/ce2b1069-daf1-422d-9910-77da05ee8bc4)

There are lot of steps to set up a basic system where the expected result is to add a node and _see something_

Another example and much more common case:

* Collision Objects

To add a working physics body or area trigger the user needs to:
1 - Add a body/area
2 - Add a CollisionShape/Polygon child
3 - Set a Shape/Draw the polygon

These are a lot of steps, again, the basic required steps are always the same, missing parts are not clearly highlighted either, just a warning in the node is all what the user gets.
This workflow is really slow when, for example, adding lot of static bodies _with different shapes_ to fill a level.

-----

So, what about Node configuration presets?

![image](https://github.com/godotengine/godot-proposals/assets/10215987/6f77c138-1d93-43f8-88e6-4b75e3ad439c)

Having a default (by default but configurable) preset, adding nodes will do a bit more than just add a node, for particles for example it could add a the emitter, add some basic mesh and process material, there could be several presets just to help setting up the system.

For bodies and areas it could add the CollisionShape node too with a basic Shape and even a placeholder Mesh or Sprite sibling.

Many other nodes could benefit of this, and if it is customizable, users can add more presets that can work as placeholders without the need to pollute projects with temporary scenes.

**Comment by AThousandShips** [comment]
See also:
* https://github.com/godotengine/godot-proposals/issues/6057

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/pull/88647.

Combining the above proposal with automatically expanding newly created resources in the inspector should help reduce the number of clicks required by a lot.

For particles, I feel we should have a dedicated setup wizard at the top of the editor when you have a particles node selected. There are a lot of possible combinations and setups you'd want to be able to start off with.

**Reply by eon-s** [reply]
My idea of preset is so these can be even downloaded and used as generic presets, a wizard can be hard to customize and affect consistency
